source/includes/connection-snippets/x509-snippets/java-connection.java

```diff
@@ -0,0 +1,37 @@
+// begin x509 connection
+import java.util.Arrays;
+import org.bson.Document;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoCredential;
+import com.mongodb.ServerAddress;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoIterable;
+
+public class X509Connection {
+
+    public static void main(String args[]) {
+
+        System.setProperty("javax.net.ssl.trustStore", "/etc/certs/mongodb/v3/client.truststore");
+        System.setProperty("javax.net.ssl.trustStorePassword", "<your_password>");
+
+        System.setProperty("javax.net.ssl.keyStore", "/etc/certs/mongodb/v3/client.keystore");
+        System.setProperty("javax.net.ssl.keyStorePassword", "<your_password>");
+
+        try {
+            String subject = "<your_subject>"; // x509 client certificate subject
```

Comment:
Why aren't you using a connection string? Were you unable to get it to work?

Reply:
@ccho-mongodb might be able to shed some light on this, since this was his original code that I've adapted for this repo

Reply:
@mongomoe It seems related to Jeff's other comment here: `source/includes/connection-snippets/scram-snippets/java-connection.java`. I think the request was to try using a connection string (e.g. similar to other examples such as the golang one) rather than using the builder pattern.